server/src/lib/access_control/AccessController.ts

```diff
@@ -47,11 +43,25 @@ export class AccessController implements IAccessController {
     this.configuration = configuration;
   }
 
-  private isAccessAllowedInRules(rules: ACLRule[], domain: string, resource: string): AccessReturn {
+  private SelectPolicy(whitelisted: WhitelistValue, secondFactorAuth: boolean) {
+    const that = this;
+    return function (rule: ACLRule): ("allow" | "deny") {
+      if (whitelisted > WhitelistValue.NOT_WHITELISTED) {
+        const whitelistPolicy = rule.whitelist_policy || that.configuration.default_whitelist_policy;
+        if (whitelistPolicy == "deny" &&
+          whitelisted > (secondFactorAuth ? WhitelistValue.WHITELISTED_AND_AUTHENTICATED_FIRSTFACTOR : WhitelistValue.WHITELISTED))
+          return rule.policy;
```

Comment:
I don't see the point in mixing whitelist policy with default policy.
I think, if you convince me that the feature of dual policy is valuable, we should completely separate both policies as if they were completely distinct.